scripts/data/generate_data.py

```diff
@@ -173,37 +185,41 @@ def fetch_block(block_hash: str, fast: bool):
     )
 
     block["data"] = {
-        tx["txid"]: resolve_transaction(tx, previous_outputs)
+        tx["txid"]: resolve_transaction(
+            tx, previous_outputs, block["mediantime"]
```

Comment:
Ok, so you need to do some more changes, because here you should provide `current_chain_state["median_time_past"]`

You need to implement an "apply" function for chain state that would produce next_chain_state from the current_chain_state. Then you provide chain_state to the fetch_block function to get MTP.

As a bonus you'll be able to get rid of fetch_chain_state function because now you are computing it from the previous ones.